velox/functions/sparksql/tests/StringTest.cpp

```diff
@@ -171,6 +171,82 @@ TEST_F(StringTest, conv) {
   EXPECT_EQ(conv("", 10, std::nullopt), std::nullopt);
 }
 
+TEST_F(StringTest, levenshtein) {
+  const auto levenshteinTest = [&](const std::optional<std::string>& left,
+                                   const std::optional<std::string>& right,
+                                   const std::optional<int32_t>& threshold,
+                                   int32_t levenshteinExpected,
+                                   int32_t levenshteinWithThresholdExpected) {
+    EXPECT_EQ(
+        evaluateOnce<int32_t>("levenshtein(c0, c1)", left, right),
+        levenshteinExpected);
+    EXPECT_EQ(
+        evaluateOnce<int32_t>(
+            "levenshtein(c0, c1, c2)", left, right, threshold),
+        levenshteinWithThresholdExpected);
+  };
+
+  levenshteinTest("kitten", "sitting", 2, 3, -1);
```

Comment:
To improve readability and reduce duplicate testing, perhaps we can adopt a function as below and separate the tests for w/wo threshold.

```
  const auto levenshtein = [&](const std::optional<std::string>& left,
                               const std::optional<std::string>& right,
                               const std::optional<int32_t>& threshold = std::nullopt) {
    if (threshold.has_value()) {
      return evaluateOnce<int32_t>("levenshtein(c0, c1, c2)", left, right, threshold);
    }
    return evaluateOnce<int32_t>("levenshtein(c0, c1)", left, right);
  };

  EXPECT_EQ(levenshtein("kitten", "sitting"), 3);
  EXPECT_EQ(levenshtein("kitten", "sitting", 2), -1);
```

Reply:
Updated, thanks.